### ISSUE TITLE
run *.cmd from wsl shell

### ISSUE THREAD

**Issue by oHaiyang** [microsoft/WSL#4547]
Microsoft Windows [Version 10.0.18980.1]

The documents say one can [Run Windows tools from WSL](https://docs.microsoft.com/en-us/windows/wsl/interop#run-windows-tools-from-wsl), if one type `notepad.exe` into the WSL shell, the Windows application should launch. But what if I want to run a cmd script directly from WSL shell, is that possible?

For some tools like yarn and npm, they do not have `yarn.exe` or `npm.exe`, but use a `.cmd` script as their command interface.

**Comment by archilkarchava** [comment]
You can make a wrapper bash/zsh/fish/sh script for it. Something like `cmd.exe /c <yourscript>.cmd`.

**Comment by oHaiyang** [comment]
Thanks, I will have a try and post the scripts for someone may have same situation.

**Comment by therealkenc** [comment]
The use case is hard to follow here though. Yep, `cmd.exe /c` is how you would invoke a `.cmd` script from WSL, but no _Linux_ `npm` would ever do that. Conversely once you invoke `npm.exe` you are off in Windows-land, and anything in the `npm.exe` process tree after that happens in Windows. The Windows `npm.exe` will spawn whatever it needs to spawn, which might end up being some `.cmd` script; but WSL is oblivious.

But whatever the particulars of the use case, @archilkarchava has the canonical answer of how to evoke a `.cmd` script from WSL.

[Once some time back when someone else asked this question, I had wondered idly if anyone had done a `cmd.exe` interpreter look-alike in Linux. Never did find one. There is Linux [pwsh](https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell-core-on-linux?view=powershell-6), which is pretty cool.]

**Comment by oHaiyang** [comment]
@archilkarchava Thanks again.

end up with a simple shell function that works great for me:

```bash
cmd() {
  CMD=$1
  shift;
  ARGS=$@
  WIN_PWD=`wslpath -w "$(pwd)"`
  cmd.exe /c "pushd ${WIN_PWD} && ${CMD} ${ARGS}"
}
```

then I can simply call `cmd yarn` in WSL terminal like it's running yarn in Windows environment. 

@therealkenc 
yes, I want to have the yarn/npm running in Windows-land. I'm using neovim in WSL as the code editor, but running electron in windows as the runtime for my code. Running yarn/npm in Windows-land makes it download pre-built electron binary for Windows but Linux.

**Comment by aartek** [comment]
@oHaiyang's fish function equivalent to run windows scripts from fish on wsl
```
function cmd 
    set CMD $argv[1]
    set ARGS $argv[2..-1]
    set WIN_PWD (wslpath -w (pwd))
    echo $WIN_PWD
    cmd.exe /c "pushd $WIN_PWD && $CMD $ARGS"
end
```
